File: Leetcode/Math/#342-Power of Four/main.cpp
class Solution {
public:
    bool isPowerOfFour(int n) {
        if (n <= 0) {
            return false;
        }

        if (n == 1) {
            return true;
        }

        if (n % 4 != 0) {
            return false;
        }

        return isPowerOfFour(n / 4);
    }
};

File: Leetcode/Math/#342-Power of Four/main.py
class Solution(object):
    def isPowerOfFour(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num < 4:
            if num == 1: return True
            else: return False
        elif num % 4 == 0: return self.isPowerOfFour(num / 4)
        else: return False
        
